### ISSUE TITLE
Task level parameters are not validated

### ISSUE THREAD

**Issue by gintautassulskus** [ansible/ansible#28340]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

_Include_ task.

##### ANSIBLE VERSION

```
ansible 2.3.2.0
  config file =
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Jul 18 2017, 09:17:00) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### SUMMARY

_Include_ task does not validate task-level parameters. E.g. if parameter _run_once_ is misspelt, the task will be executed without a failure.
For instance, a _ping_ task throws a validation error in such a case.
  
##### STEPS TO REPRODUCE

```
- include: subtasks.yml
  run_once: yes
  non_existent_parameter: yes!
```

##### EXPECTED RESULTS

ERROR! 'non_existent_parameter' is not a valid attribute for a Task

##### ACTUAL RESULTS

The task has been executed.

**Comment by giovannisciortino** [comment]
Also "include_tasks" has the same issue in ansible version 2.4.0.
 
Step to reproduce:
```
    - include_tasks: subtasks.yml
      run_once: yes
      non_existent_parameter: yes!
```

I'm trying to work on this issue

**Comment by giovannisciortino** [comment]
Further analysis on this issue:

- The parameter "non_existent_parameter" is parsed as a variable for the included tasks as documented in http://docs.ansible.com/ansible/playbooks_roles.html#task-include-files-and-
encouraging-reuse for this reason the check of task level parameters is not validated on it.

- If the task described in this issue is inserted in a playbook using ansible 2.3.0 the following deprecation warning is showed: 

```
[DEPRECATION WARNING]: Specifying include variables at the top-level of the task is deprecated. Please see:
http://docs.ansible.com/ansible/playbooks_roles.html#task-include-files-and-
encouraging-reuse

 for currently supported syntax regarding included files and variables.
This feature will be removed in a future release. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
```

- In ansible 2.4 the deprecation warning message is a bit different, it contains the phrase "This feature will be removed in version 2.7."

Below I report a full example:
```
$ ansible --version
ansible 2.3.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, May 10 2017, 20:04:28) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]

$ cat /tmp/subtasks.yml 
- debug: msg="{{non_existent_parameter}}"

$ cat /tmp/playbook.yml 
- hosts: localhost
  gather_facts: no
  tasks:
    - include: subtasks.yml
      run_once: yes
      non_existent_parameter: yes!

$ ansible-playbook /tmp/playbook.yml 
[DEPRECATION WARNING]: Specifying include variables at the top-level of the task is deprecated. Please see:
http://docs.ansible.com/ansible/playbooks_roles.html#task-include-files-and-encouraging-reuse

 for currently supported syntax regarding included files and variables.
This feature will be removed in a future release. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.

PLAY [localhost] *****************************************************************************************************************************************************************************

TASK [debug] *********************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false, 
    "msg": "yes!"
}

PLAY RECAP ***********************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0   
```

Based on the reasoning described above, I don't think that this is a issue

**Comment by bcoca** [comment]
closing as per above

**Comment by gintautassulskus** [comment]
@giovannisciortino, thanks for pointing out that based on the documentation this is not an issue.
I would like to argue that the current approach introduces ambiguities through unnecessary syntax overloading. My arguments:
 
1. Tasks use "vars:" block to declare variables whereas _include:_ does not require it. This creates a special case where _include:_ is called a _task_ but is treated differently from other tasks. 
2. Parameter "vars:" helps to clearly distinguish task level parameters from variables. Without the parameter, it becomes more difficult to read and debug the scripts. 
I was stuck for a good hour trying to debug a problem which turned out to be a mistyped conditional parameter _when:_ for _include:_. The mistyped parameter was treated as a variable rather than a condition. This is **dangerous** because it allows a mistype to quietly change the behaviour of the script rather than fail.
3. Without "vars:" block, variable names might clash with parameter names, e.g. both a variable and a parameter can be named "with_items". This creates the need to document and maintain a list of reserved keywords.

To summarise, omitting the requirement for the "vars:" creates a special case that requires case-specific documentation, maintenance, complicates debugging and might even silently change the behaviour of a script.
The only benefit I see in this is that we can omit one line of "vars:".

**Comment by bcoca** [comment]
@gintautassulskus due to those problems is why we deprecated the 'feature' .. which predates 'task vars keyword`, which is the correct way of doing this now.
